packages/core/src/components/ProgressBar.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  defineComponent,
+  h,
+  inject,
+  onMounted,
+  onUpdated,
+  onUnmounted,
+  computed,
+} from '@vue/runtime-core'
+import type { PropType } from '@vue/runtime-core'
+import { colorize } from '../renderer/textColor'
+import type { ForegroundColorProp } from '../renderer/textColor'
+import { scheduleUpdateSymbol } from '../injectionSymbols'
+
+const FIGURES = {
+  basic: '█',
+  shade: '▓',
+} as const
+
+type FigureType = keyof typeof FIGURES
+
+export const TuiProgressBar = defineComponent({
+  name: 'TuiProgressBar',
+
+  props: {
+    color: {
+      required: false,
+      default: 'blue',
+      type: String as PropType<ForegroundColorProp>,
+    },
+    bgColor: {
+      required: false,
+      default: 'white',
+      type: String as PropType<ForegroundColorProp>,
+    },
+    barWidth: {
```

Comment:
I think just width is fine since the bar's width is what one would think of by default